File: app/src/main/java/com/example/james/rms/CommonProfile/MyAppGlideModule.java
package com.example.james.rms.CommonProfile;

import android.content.Context;


import com.bumptech.glide.GlideBuilder;
import com.bumptech.glide.annotation.GlideModule;
import com.bumptech.glide.load.DecodeFormat;
import com.bumptech.glide.module.AppGlideModule;

/**
 * Created by Jamie on 17/6/2017.
 */

@GlideModule
public class MyAppGlideModule extends AppGlideModule{

    @Override
    public void applyOptions(Context context, GlideBuilder builder) {
        // Apply options to the builder here.
        builder.setDecodeFormat(DecodeFormat.PREFER_ARGB_8888);
    }

}

File: app/src/main/java/com/example/james/rms/glide/MyAppGlideModule.kt
package com.example.james.rms.glide

import android.content.Context
import com.bumptech.glide.Glide


import com.bumptech.glide.GlideBuilder
import com.bumptech.glide.Registry
import com.bumptech.glide.annotation.GlideModule
import com.bumptech.glide.integration.okhttp3.OkHttpUrlLoader
import com.bumptech.glide.load.DecodeFormat
import com.bumptech.glide.load.model.GlideUrl
import com.bumptech.glide.module.AppGlideModule
import com.bumptech.glide.request.RequestOptions
import java.io.InputStream

/**
 * Created by Jamie on 17/6/2017.
 */

@GlideModule
class MyAppGlideModule : AppGlideModule() {

    override fun applyOptions(context: Context, builder: GlideBuilder) {
        // Apply options to the builder here.
        builder.setDefaultRequestOptions(RequestOptions().format(DecodeFormat.PREFER_ARGB_8888))
    }

    override fun isManifestParsingEnabled(): Boolean = false
    override fun registerComponents(context: Context?, registry: Registry?) {
        super.registerComponents(context, registry)
        // Replace the http connection with okhttp
        registry!!.replace(GlideUrl::class.java, InputStream::class.java, OkHttpUrlLoader.Factory())
    }


}
